src/protocol/ipa/mod.rs

```diff
@@ -155,24 +156,60 @@ impl<F: Field + Sized, T: Arithmetic<F>> Resharable<F> for IPAModulusConvertedIn
     }
 }
 
+/// Describes a request to execute IPA protocol. Consists of inputs for the computation
+/// and query parameters
+pub struct Query<'a, F: Field, MK: BitArray = MatchKey, BK: BitArray = BreakdownKey> {
+    input: &'a [IPAInputRow<F, MK, BK>],
+    /// Maximum contribution allowed per match key/user.
+    per_user_credit_cap: u32,
+    /// ?
```

Comment:
IPA queries report a total per "breakdown key". These keys are specified by the report collector, and secret-shared along with "source reports". All breakdown keys should be between [0, max_breakdown_key).

Reply:
I think that num_breakdown_keys would be a better name for that then.